models/spring-ai-bedrock/src/main/java/org/springframework/ai/bedrock/titan/BedrockTitanChatModel.java

```diff
@@ -78,12 +78,17 @@ public Flux<ChatResponse> stream(Prompt prompt) {
 			ChatGenerationMetadata chatGenerationMetadata = null;
 			if (chunk.amazonBedrockInvocationMetrics() != null) {
 				String completionReason = chunk.completionReason().name();
-				chatGenerationMetadata = ChatGenerationMetadata.from(completionReason,
-						chunk.amazonBedrockInvocationMetrics());
+				chatGenerationMetadata = ChatGenerationMetadata.builder()
+					.finishReason(completionReason)
+					.metadata("usage", chunk.amazonBedrockInvocationMetrics())
```

Comment:
Should be "metrics" ?

Reply:
The class BedrockUsage has  `AmazonBedrockInvocationMetrics getUsage()` so maybe it is the other way around, the class name should change?  Prob not that important as we will remove this on the switch to bedrock converse.